stairway/src/main/java/bio/terra/stairway/impl/FlightFilterAccess.java

```diff
@@ -289,19 +278,66 @@ private void storeFlightPredicateValue(
    */
   @VisibleForTesting
   String makeInputPredicateSql(FlightFilterPredicate predicate) {
-    return "(I.key = '"
+    String valuePredicate = "";
+    if (predicate.getDatatype() == Datatype.NULL) {
+      return "(EXISTS (SELECT 0 FROM flightinput I WHERE F.flightid = I.flightid AND I.key = "
+          + "'"
+          + predicate.getKey()
+          + "' AND I.value IS NULL) "
+          + "OR NOT EXISTS (SELECT 0 FROM flightinput I WHERE F.flightid = I.flightid AND I.key = "
+          + "'"
+          + predicate.getKey()
+          + "'))";
+    } else if (predicate.getDatatype() == Datatype.LIST) {
+      valuePredicate =
+          "I.value "
+              + predicate.getOp().getSql()
+              + " (SELECT JSON_ARRAY_ELEMENTS_TEXT(CAST(:"
+              + predicate.getParameterName()
+              + " AS JSON)))";
+    } else {
+      valuePredicate = "I.value" + predicate.getOp().getSql() + ":" + predicate.getParameterName();
+    }
+
+    return "EXISTS (SELECT 0 FROM flightinput I WHERE F.flightid = I.flightid AND I.key = "
+        + "'"
         + predicate.getKey()
-        + "' AND I.value"
-        + predicate.getOp().getSql()
-        + ":"
-        + predicate.getParameterName()
+        + "' AND "
+        + valuePredicate
         + ")";
   }
 
+  private void storeInputPredicateValue(
+      FlightBooleanOperationExpression booleanExpression, NamedParameterPreparedStatement statement)
+      throws SQLException, JsonProcessingException {
+
+    for (FlightBooleanOperationExpression expression : booleanExpression.getExpressions()) {
+      storeInputPredicateValue(expression, statement);
+    }
+    if (booleanExpression.getBasePredicate() != null) {
+      storeInputPredicateValue(booleanExpression.getBasePredicate(), statement);
+    }
+  }
+
   private void storeInputPredicateValue(
       FlightFilterPredicate predicate, NamedParameterPreparedStatement statement)
       throws SQLException, JsonProcessingException {
-    String jsonValue = StairwayMapper.getObjectMapper().writeValueAsString(predicate.getValue());
+    if (predicate.getDatatype() == Datatype.NULL) {
+      return;
+    }
+    String jsonValue;
+    if (predicate.getDatatype() == Datatype.LIST) {
+      List<String> values = new ArrayList<>();
+      for (Object value : ((List<?>) predicate.getValue())) {
+        // Need to double encode in order for strings to match
+        values.add(
+            StairwayMapper.getObjectMapper()
+                .writeValueAsString(StairwayMapper.getObjectMapper().writeValueAsString(value)));
+      }
+      jsonValue = "[" + StringUtils.join(values, ",") + "]";
```

Comment:
`StringUtils` is unnecessary here
```suggestion
      jsonValue = "[" + String.join(",", values) + "]";
```